File: imagelab-backend/app/models/batch.py
from typing import Literal

from pydantic import BaseModel


class BatchResultItem(BaseModel):
    filename: str
    success: bool
    output_filename: str | None = None
    error: str | None = None


class BatchJobSummary(BaseModel):
    job_id: str
    status: Literal["pending", "processing", "completed", "failed"]
    total_files: int
    processed_files: int
    success_count: int
    failure_count: int
    created_at: float
    updated_at: float
    duration_seconds: float
    results: list[BatchResultItem] = []


class BatchJobResponse(BaseModel):
    job_id: str
    status: str
    total_files: int

File: imagelab-frontend/src/types/batch.ts
export interface BatchResultItem {
  filename: string;
  success: boolean;
  output_filename: string | null;
  error: string | null;
}

export interface BatchJobSummary {
  job_id: string;
  status: "pending" | "processing" | "completed" | "failed";
  total_files: number;
  processed_files: number;
  success_count: number;
  failure_count: number;
  created_at: number;
  updated_at: number;
  duration_seconds: number;
  results: BatchResultItem[];
}

export interface BatchJobResponse {
  job_id: string;
  status: string;
  total_files: number;
}
